Makefile

```diff
@@ -94,11 +95,31 @@ e2e-tests-es: prepare-e2e-tests es
 	@echo Running Elasticsearch end-to-end tests...
 	@STORAGE_NAMESPACE=$(STORAGE_NAMESPACE) go test -tags=elasticsearch ./test/e2e/... -kubeconfig $(KUBERNETES_CONFIG) -namespacedMan ../../deploy/test/namespace-manifests.yaml -globalMan ../../deploy/crds/jaegertracing_v1_jaeger_crd.yaml -root .
 
+.PHONY: e2e-tests-self-provisioned-es
+e2e-tests-self-provisioned-es: prepare-e2e-tests deploy-es-operator
+	@echo Running Self provisioned Elasticsearch end-to-end tests...
+	@go test -tags=self_provisioned_elasticsearch ./test/e2e/... -kubeconfig $(KUBERNETES_CONFIG) -namespacedMan ../../deploy/test/namespace-manifests.yaml -globalMan ../../deploy/crds/jaegertracing_v1_jaeger_crd.yaml -root .
+
 .PHONY: run
 run: crd
 	@rm -rf /tmp/_cert*
 	@bash -c 'trap "exit 0" INT; OPERATOR_NAME=${OPERATOR_NAME} KUBERNETES_CONFIG=${KUBERNETES_CONFIG} WATCH_NAMESPACE=${WATCH_NAMESPACE} go run -ldflags ${LD_FLAGS} main.go start'
 
+.PHONY: set-max-map-count
+set-max-map-count:
+	@minishift ssh -- 'sudo sysctl -w vm.max_map_count=262144' > /dev/null 2>&1 || true
+
+.PHONY: deploy-es-operator
+deploy-es-operator: set-max-map-count
```

Comment:
@josefkarasek could you please have a look and verify we are deploying es-operator correctly? Especially if the p8s crds are required.

Reply:
Yes, those CRDs are required.
If cluster is provisioned by openshift/installer, the openshift-monitoring CRDs should be created by the installer.
In you use minishift you probably have to create them manually.